### DISCUSSION TITLE
Clean up machine actor with huge contexts after it’s finished?

### DISCUSSION THREAD

**Discussion by domvo** [statelyai/xstate#5068] [Q&A]
We’re using xState for a RAG document ingestion pipeline (btw, **amazing** use case for it). 

Naturally we’re handling loads of data here and I put huge chunks of data into the context of a machine. 

For every document ingestion I create a new actor, so over time loads of actors are created with huge contexts. 

Now I’m wondering whether it makes sense to clean up the context manually when arriving in my final states to free up the memory. Or is this handled by xState internally?

**Comment by Andarist** [comment]
The context of a finished actor won't be cleared but if u dispose the child actor from its parent then GC should kick in as usual.
